src/controllers/auth.js

```diff
@@ -1,2 +1,27 @@
 const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
 const UserModel = require('../models/user');
+
+const saveUserToTokenAndCookie = (req, res) => {
+  const { name, email, providerId, profilePicture } = req.user;
+  const payload = {
+    name,
+    email,
+    providerId,
+    avatar: profilePicture,
+  };
+  const token = jwt.sign(payload, process.env.SECRET_KEY, {
+    expiresIn: '14d',
+  });
+  res.cookie('token', token, {
+    httpOnly: true,
+    signed: true,
+    maxAge: 1000 * 60 * 60 * 24 * 14,
+  });
+
+  res.redirect('/');
```

Comment:
you don't need to redirect it

Reply:
I tried it without the redirect and it's not going anywhere...

Reply:
I thought it would be frontend's responsibility to redirect. Should we add them really? Halit if you are seeing this comment please answer xd

Reply:
I don't think so. usually the redirection is done in the middleware even as a successRedirect option, but using that would prevent the code from reaching the controller and thus we won't be able to save the data we want to the JWT. But if you don't use redirect the request won't be resolved and the code won't get out of the callback function.
![image](https://user-images.githubusercontent.com/98318641/182002752-643881e2-a8be-4253-b40c-c59d9cf0e427.png)